examples/inspection/plot_linear_model_coefficient_interpretation.py

```diff
@@ -0,0 +1,602 @@
+y_pred = model.predict(X_test)
+mae = median_absolute_error(y_test, y_pred)
+string_score += '\nMAE on testing set: {0:.2f} $/hour'.format(mae)
+fig, ax = plt.subplots(figsize=(5, 5))
+plt.scatter(y_test, y_pred)
+ax.plot([0, 1], [0, 1], transform=ax.transAxes, ls="--", c="red")
+plt.text(3, 20, string_score)
+plt.title('Ridge model, small regularization')
+plt.ylabel('Model predictions')
+plt.xlabel('Truths')
+plt.xlim([0, 27])
+plt.ylim([0, 27])
+
+##############################################################################
+# The model learnt is far from being a good model making accurate predictions:
+# this is obvious when looking at the plot above, where good predictions
+# should lie on the red line.
+# In the following section, we will interpret the coefficients of the model.
+# While we do so, we should keep in mind that any conclusion we way draw will
+# be about
+# the model that we build, rather than about the true (real-world) generative
+# process of the data.
+#
+# Interpreting coefficients: scale matters
+# ---------------------------------------------
+#
+# First of all, we can take a look to the values of the coefficients of the
+# regressor we have fitted.
+
+feature_names = (model.named_steps['columntransformer']
+                      .named_transformers_['onehotencoder']
+                      .get_feature_names(input_features=categorical_columns))
+feature_names = np.concatenate(
+    [feature_names, numerical_columns])
+
+coefs = pd.DataFrame(
+    model.named_steps['transformedtargetregressor'].regressor_.coef_,
+    columns=['Coefficients'], index=feature_names
+)
+
+coefs
+
+##############################################################################
+# The AGE coefficient is expressed in
+# :math:`$/hours/(living\ years)` while the EDUCATION one is expressed
+# in :math:`$/hours/(years\ of\ education)`.
+# This representation of the coefficients has the advantage of making clear
+# the practical predictions of the model:
+# an increase of :math:`1` year in AGE means a decrease of :math:`0.030867$`,
+# while an increase of :math:`1` year in EDUCATION means an increase of
+# :math:`0.054699$`.
+# On the other hand, categorical variables (as UNION or SEX) are adimensional
+# numbers taking the value either of 0 or 1. Their coefficients are expressed
+# in :math:`$/hours`. Then, we cannot compare the magnitude of different
+# coefficients since the features have different natural scales, and hence
+# value ranges, because of their different unit of measure.
+# This is more evident if we plot the coefficients.
+
+coefs.plot(kind='barh', figsize=(9, 7))
+plt.title('Ridge model, small regularization')
+plt.axvline(x=0, color='.5')
+plt.subplots_adjust(left=.3)
+
+###############################################################################
+# Indeed, from the plot above the most important factor in determining WAGE
+# appears to be the
+# variable UNION, even if it is plausible that variables like EXPERIENCE
```

Comment:
I'm seeing the biggest bars on OCCUPATION variables. Is that right?

Reply:
You are right. Sorry: I have changed the encoder in the preprocessing and didn't notice the change in the coefficients. I should have dropped one column in the binary variables. I'm attaching the old plot for completeness.
![oldplot](https://user-images.githubusercontent.com/1662261/75250779-6c04e000-57d9-11ea-827f-5bef84129e0a.png)

Reply:
Fixed using `drop='binary'` in OneHotEncoder.